src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -936,34 +1059,80 @@ impl FromIntoArrow for Time {
     }
 }
 
-impl FromIntoArrow for Timestamp {
+impl FromIntoArrowWithUnit for Timestamp {
     type ArrowType = i64;
+    type TimestampType = TimeUnit;
 
-    fn from_arrow(value: Self::ArrowType) -> Self {
-        Timestamp(
-            DateTime::from_timestamp((value / 1_000_000) as _, (value % 1_000_000 * 1000) as _)
+    fn from_arrow_with_unit(value: Self::ArrowType, time_unit: Self::TimestampType) -> Self {
+        match time_unit {
+            TimeUnit::Second => {
+                Timestamp(DateTime::from_timestamp(value as _, 0).unwrap().naive_utc())
+            }
+            TimeUnit::Millisecond => Timestamp(
+                DateTime::from_timestamp((value / 1_000) as _, (value % 1_000 * 1000000) as _)
```

Comment:
I think you can use [DateTime::from_timestamp_millis](https://doc.servo.org/chrono/struct.DateTime.html#method.from_timestamp_millis) here.